ETL (Extracción, Transformación y Carga) del dataset internet

Carga de datos

In [1]:
import pandas as pd
import numpy as np


Se analiza el dataset desde el enlace:"https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx": descargar y abrir en Python con Pandas y openpyxl para explorar el contenido. 


In [2]:
# Cargar el archivo Excel
df = pd.read_excel("https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx", engine="openpyxl")

# Mostrar las primeras filas para entender el contenido
print(df.head())
print(df.info())
print(df.describe())

        Partido   Localidad  link Indec Velocidad (Mbps)  Provincia  Accesos
0  BUENOS AIRES  25 de Mayo  25 de Mayo          6854100       0.00      1.0
1  BUENOS AIRES  25 de Mayo  25 de Mayo          6854100       0.50      2.0
2  BUENOS AIRES  25 de Mayo  25 de Mayo          6854100       0.75     19.0
3  BUENOS AIRES  25 de Mayo  25 de Mayo          6854100       3.00     85.0
4  BUENOS AIRES  25 de Mayo  25 de Mayo          6854100       3.50    145.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18864 entries, 0 to 18863
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Partido           18864 non-null  object 
 1   Localidad         18864 non-null  object 
 2   link Indec        18863 non-null  object 
 3   Velocidad (Mbps)  18864 non-null  object 
 4   Provincia         18864 non-null  float64
 5   Accesos           18857 non-null  float64
dtypes: float64(2), object(4)
memory usage: 884.4+ KB


Verificar el numero de hojas que tiene el archivo y cuáles son sus nombres

In [3]:
# Carga el archivo sin leer ninguna hoja en específico
archivo_excel = pd.ExcelFile("https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx")

# Mostrar el número de hojas y sus nombres
numero_hojas = len(archivo_excel.sheet_names)
nombres_hojas = archivo_excel.sheet_names

print(f"Número de hojas: {numero_hojas}")
print(f"Nombres de hojas: {nombres_hojas}")

Número de hojas: 15
Nombres de hojas: ['Acc_vel_loc_sinrangos', 'Velocidad_sin_Rangos', 'Accesos_tecnologia_localidad', 'Velocidad % por prov', 'Totales VMD', 'Totales Accesos Por Tecnología', 'Accesos Por Tecnología', 'Penetración-poblacion', 'Penetracion-hogares', 'Penetracion-totales', 'Totales Accesos por rango', 'Accesos por rangos', 'Dial-BAf', 'Totales Dial-BAf', 'Ingresos ']


Analizar cada hoja individualmente y decidir cuáles usar para el análisis

FALTA DE 'Velocidad_sin_Rangos' Y DE 'Totales Dial-BAf' ESTOY AQUIIII

Analisis y ETL de la Hoja1: Acc_vel_loc_sinrangos

In [5]:
# Acc_vel_loc_sinrangos
df_acc_vel_loc = pd.read_excel(archivo_excel, sheet_name='Acc_vel_loc_sinrangos')
print("Hoja: Acc_vel_loc_sinrangos")
print(df_acc_vel_loc.head())
print(df_acc_vel_loc.info())
print(df_acc_vel_loc.describe())

Hoja: Acc_vel_loc_sinrangos
        Partido   Localidad  link Indec Velocidad (Mbps)  Provincia  Accesos
0  BUENOS AIRES  25 de Mayo  25 de Mayo          6854100       0.00      1.0
1  BUENOS AIRES  25 de Mayo  25 de Mayo          6854100       0.50      2.0
2  BUENOS AIRES  25 de Mayo  25 de Mayo          6854100       0.75     19.0
3  BUENOS AIRES  25 de Mayo  25 de Mayo          6854100       3.00     85.0
4  BUENOS AIRES  25 de Mayo  25 de Mayo          6854100       3.50    145.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18864 entries, 0 to 18863
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Partido           18864 non-null  object 
 1   Localidad         18864 non-null  object 
 2   link Indec        18863 non-null  object 
 3   Velocidad (Mbps)  18864 non-null  object 
 4   Provincia         18864 non-null  float64
 5   Accesos           18857 non-null  float64
dtypes: float64(2), object

In [18]:
# Paso 1: Limpiar la columna 'Velocidad (Mbps)' (convertir a numérico)
df_acc_vel_loc['Velocidad (Mbps)'] = pd.to_numeric(df_acc_vel_loc['Velocidad (Mbps)'], errors='coerce')

# Paso 2: Manejar valores nulos
# Rellenar nulos en 'link Indec' con un valor predeterminado, por ejemplo, 'Desconocido'
df_acc_vel_loc['link Indec'] = df_acc_vel_loc['link Indec'].fillna('Desconocido')
# Rellenar nulos en 'Accesos' con 0, o podrías usar la mediana o el promedio
df_acc_vel_loc['Accesos'] = df_acc_vel_loc['Accesos'].fillna(0)

# Paso 3: Eliminar o manejar valores atípicos en 'Accesos'
# Suponiendo que los valores negativos no son válidos para "Accesos"
df_acc_vel_loc = df_acc_vel_loc[df_acc_vel_loc['Accesos'] >= 0]

# Paso 4: Verificar resultados
print("\nInformación después de la limpieza:")
print(df_acc_vel_loc.info())
print(df_acc_vel_loc.describe())

# Paso 5: Guardar el DataFrame limpio 
df_acc_vel_loc.to_csv('Acc_vel_loc_sinrangos_limpio.csv', index=False)



Información después de la limpieza:
<class 'pandas.core.frame.DataFrame'>
Index: 18860 entries, 0 to 18863
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Partido           18860 non-null  object 
 1   Localidad         18860 non-null  object 
 2   link Indec        18860 non-null  object 
 3   Velocidad (Mbps)  18726 non-null  float64
 4   Provincia         18860 non-null  float64
 5   Accesos           18860 non-null  float64
dtypes: float64(3), object(3)
memory usage: 1.0+ MB
None
       Velocidad (Mbps)     Provincia        Accesos
count      1.872600e+04  18860.000000   18860.000000
mean       3.269266e+07     66.864317     613.102333
std        2.960939e+07    178.996081    6282.692560
min        2.000010e+06      0.000000       0.000000
25%        6.560010e+06      3.000000       3.000000
50%        1.416107e+07     10.000000      26.000000
75%        5.809102e+07     30.000000     180.000000
max

Analisis y ETL de la Hoja 2: Velocidad_sin_Rangos

In [5]:
# Velocidad_sin_Rangos
df_acc_vel_loc = pd.read_excel(archivo_excel, sheet_name='Velocidad_sin_Rangos')
print("Hoja: Velocidad_sin_Rangos")
print(df_acc_vel_loc.head())
print(df_acc_vel_loc.info())
print(df_acc_vel_loc.describe())

Hoja: Velocidad_sin_Rangos
    Año  Trimestre     Provincia  Velocidad  Accesos
0  2024          2  BUENOS AIRES       75.0     1062
1  2024          2  BUENOS AIRES       59.0       59
2  2024          2  BUENOS AIRES      480.0        5
3  2024          2  BUENOS AIRES        3.5    41735
4  2024          2  BUENOS AIRES       18.0     1042
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18884 entries, 0 to 18883
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Año        18884 non-null  int64  
 1   Trimestre  18884 non-null  int64  
 2   Provincia  18884 non-null  object 
 3   Velocidad  18874 non-null  float64
 4   Accesos    18884 non-null  int64  
dtypes: float64(1), int64(3), object(1)
memory usage: 737.8+ KB
None
                Año     Trimestre     Velocidad       Accesos
count  18884.000000  18884.000000  18874.000000  1.888400e+04
mean    2021.017104      2.477653     96.171156  1.412328e+04
std        1

In [7]:
# Ruta del archivo Excel
file_path = "https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx"  # ruta de tu archivo
# Cargar la hoja específica
df = pd.read_excel(file_path, sheet_name="Velocidad_sin_Rangos")

In [9]:
# Opción 2: Eliminar filas con valores nulos
df.dropna(subset=["Velocidad"], inplace=True)


In [10]:
# Calcular el percentil 99
p99 = df["Velocidad"].quantile(0.99)

# Filtrar valores anómalos
df = df[df["Velocidad"] <= p99]

In [11]:
# Agrupación
df_grouped = df.groupby(["Año", "Trimestre", "Provincia"]).agg(
    Velocidad_Promedio=("Velocidad", "mean"),
    Total_Accesos=("Accesos", "sum")
).reset_index()

In [12]:
# Renombrar columnas
df_grouped.rename(columns={
    "Año": "anio",
    "Trimestre": "trimestre",
    "Provincia": "provincia",
    "Velocidad_Promedio": "velocidad_promedio",
    "Total_Accesos": "total_accesos"
}, inplace=True)

In [13]:
# Reducir tamaño de las columnas
df_grouped["anio"] = df_grouped["anio"].astype("int16")
df_grouped["trimestre"] = df_grouped["trimestre"].astype("int8")
df_grouped["total_accesos"] = df_grouped["total_accesos"].astype("int32")

In [14]:
# Guardar como CSV
df_grouped.to_csv("velocidad_sin_rango_limpio.csv", index=False)


Analisis y ETL de la Hoja1: Accesos_tecnologia_localidad

In [19]:
# Accesos_tecnologia_localidad
df_accesos_tecnologia_localidad = pd.read_excel(archivo_excel, sheet_name='Accesos_tecnologia_localidad')
print("\nHoja: Accesos_tecnologia_localidad")
print(df_accesos_tecnologia_localidad.head())
print(df_accesos_tecnologia_localidad.info())
print(df_accesos_tecnologia_localidad.describe())


Hoja: Accesos_tecnologia_localidad
      Provincia     Partido   Localidad    Tecnologia Link Indec  Accesos
0  BUENOS AIRES  25 de Mayo  25 de Mayo          ADSL    6854100    755.0
1  BUENOS AIRES  25 de Mayo  25 de Mayo    CABLEMODEM    6854100   4600.0
2  BUENOS AIRES  25 de Mayo  25 de Mayo  FIBRA OPTICA    6854100      2.0
3  BUENOS AIRES  25 de Mayo  25 de Mayo     SATELITAL    6854100    742.0
4  BUENOS AIRES  25 de Mayo  25 de Mayo      WIRELESS    6854100    727.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7753 entries, 0 to 7752
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Provincia   7753 non-null   object 
 1   Partido     7753 non-null   object 
 2   Localidad   7753 non-null   object 
 3   Tecnologia  7753 non-null   object 
 4   Link Indec  7753 non-null   object 
 5   Accesos     7747 non-null   float64
dtypes: float64(1), object(5)
memory usage: 363.6+ KB
None
            Accesos
count  

In [21]:
# Cargar la hoja
archivo_excel = 'https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx'  # ruta al archivo
df_accesos_tecnologia = pd.read_excel(archivo_excel, sheet_name='Accesos_tecnologia_localidad')

# Mostrar información inicial
print("Información inicial del DataFrame:")
print(df_accesos_tecnologia.info())
print(df_accesos_tecnologia.describe())

# Paso 1: Limpiar la columna 'Accesos' (convertir a numérico)
df_accesos_tecnologia['Accesos'] = pd.to_numeric(df_accesos_tecnologia['Accesos'], errors='coerce')

# Paso 2: Manejar valores nulos
# Rellenar nulos en 'Accesos' con 0
df_accesos_tecnologia['Accesos'] = df_accesos_tecnologia['Accesos'].fillna(0)

# Paso 3: Eliminar o manejar valores atípicos en 'Accesos'
# Suponiendo que los valores negativos no son válidos para "Accesos"
df_accesos_tecnologia = df_accesos_tecnologia[df_accesos_tecnologia['Accesos'] >= 0]

# Paso 4: Verificar resultados
print("\nInformación después de la limpieza:")
print(df_accesos_tecnologia.info())
print(df_accesos_tecnologia.describe())

# Paso 5: Guardar el DataFrame limpio
df_accesos_tecnologia.to_csv('Accesos_tecnologia_localidad_limpio.csv', index=False)


Información inicial del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7753 entries, 0 to 7752
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Provincia   7753 non-null   object 
 1   Partido     7753 non-null   object 
 2   Localidad   7753 non-null   object 
 3   Tecnologia  7753 non-null   object 
 4   Link Indec  7753 non-null   object 
 5   Accesos     7747 non-null   float64
dtypes: float64(1), object(5)
memory usage: 363.6+ KB
None
            Accesos
count  7.747000e+03
mean   1.491688e+03
std    1.520174e+04
min    0.000000e+00
25%    3.000000e+00
50%    5.900000e+01
75%    4.005000e+02
max    1.144781e+06

Información después de la limpieza:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7753 entries, 0 to 7752
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Provincia   7753 non-null   object 
 1   Partido     7753 non-

Analisis y ETL de la Hoja1: Velocidad % por prov

In [7]:
# Velocidad % por prov
df_velocidad_por_prov = pd.read_excel(archivo_excel, sheet_name='Velocidad % por prov')
print("\nHoja: Velocidad % por prov")
print(df_velocidad_por_prov.head())
print(df_velocidad_por_prov.info())
print(df_velocidad_por_prov.describe())


Hoja: Velocidad % por prov
    Año  Trimestre        Provincia  Mbps (Media de bajada)
0  2024          2     Buenos Aires                  157.41
1  2024          2  Capital Federal                  233.01
2  2024          2        Catamarca                   97.38
3  2024          2            Chaco                  107.76
4  2024          2           Chubut                   21.67
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1008 entries, 0 to 1007
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     1008 non-null   int64  
 1   Trimestre               1008 non-null   int64  
 2   Provincia               1008 non-null   object 
 3   Mbps (Media de bajada)  1008 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 31.6+ KB
None
               Año    Trimestre  Mbps (Media de bajada)
count  1008.000000  1008.000000             1008.000000
mean   2

In [22]:
# Cargar la hoja
archivo_excel = 'https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx'  # ruta al archivo
df_velocidad_porprov = pd.read_excel(archivo_excel, sheet_name='Velocidad % por prov')

# Mostrar información inicial
print("Información inicial del DataFrame:")
print(df_velocidad_porprov.info())
print(df_velocidad_porprov.describe())

# Paso 1: Asegurarse que 'Mbps (Media de bajada)' sea numérico
df_velocidad_porprov['Mbps (Media de bajada)'] = pd.to_numeric(df_velocidad_porprov['Mbps (Media de bajada)'], errors='coerce')

# Paso 2: Manejar valores nulos
# Rellenar nulos en 'Mbps (Media de bajada)' con la mediana o un valor adecuado
df_velocidad_porprov['Mbps (Media de bajada)'] = df_velocidad_porprov['Mbps (Media de bajada)'].fillna(df_velocidad_porprov['Mbps (Media de bajada)'].median())

# Paso 3: Eliminar o manejar valores atípicos en 'Mbps (Media de bajada)'
# Por ejemplo, si la velocidad es mayor a un límite razonable (por ejemplo, 500 Mbps), se puede considerar un valor atípico
df_velocidad_porprov = df_velocidad_porprov[df_velocidad_porprov['Mbps (Media de bajada)'] <= 500]

# Paso 4: Verificar resultados
print("\nInformación después de la limpieza:")
print(df_velocidad_porprov.info())
print(df_velocidad_porprov.describe())

# Paso 5: Guardar el DataFrame limpio
df_velocidad_porprov.to_csv('Velocidad_por_provincia_limpio.csv', index=False)


Información inicial del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1008 entries, 0 to 1007
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     1008 non-null   int64  
 1   Trimestre               1008 non-null   int64  
 2   Provincia               1008 non-null   object 
 3   Mbps (Media de bajada)  1008 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 31.6+ KB
None
               Año    Trimestre  Mbps (Media de bajada)
count  1008.000000  1008.000000             1008.000000
mean   2018.761905     2.452381               24.720245
std       3.039439     1.117574               33.660764
min    2014.000000     1.000000                2.573729
25%    2016.000000     1.000000                4.866698
50%    2019.000000     2.000000               10.089315
75%    2021.000000     3.000000               29.875292
max    2024.000000     4.

Analisis y ETL de la Hoja1: Totales VMD  

In [8]:
# Totales VMD
df_totales_vmd = pd.read_excel(archivo_excel, sheet_name='Totales VMD')
print("\nHoja: Totales VMD")
print(df_totales_vmd.head())
print(df_totales_vmd.info())
print(df_totales_vmd.describe())


Hoja: Totales VMD
    Año  Trimestre  Mbps (Media de bajada)    Trimestre.1
0  2024          2                  139.25   Abr-Jun 2024
1  2024          1                  139.15   Ene-Mar 2024
2  2023          4                  139.04   Oct-Dic 2023
3  2023          3                  129.67  Jul-Sept 2023
4  2023          2                  123.95   Abr-Jun 2023
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     42 non-null     int64  
 1   Trimestre               42 non-null     int64  
 2   Mbps (Media de bajada)  42 non-null     float64
 3   Trimestre.1             42 non-null     object 
dtypes: float64(1), int64(2), object(1)
memory usage: 1.4+ KB
None
               Año  Trimestre  Mbps (Media de bajada)
count    42.000000  42.000000               42.000000
mean   2018.761905   2.452381         

In [23]:
# Cargar la hoja
archivo_excel = 'https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx'  # ruta al archivo
df_totales_vmd = pd.read_excel(archivo_excel, sheet_name='Totales VMD')

# Mostrar información inicial
print("Información inicial del DataFrame:")
print(df_totales_vmd.info())
print(df_totales_vmd.describe())

# Paso 1: Asegurar que 'Mbps (Media de bajada)' sea numérico
df_totales_vmd['Mbps (Media de bajada)'] = pd.to_numeric(df_totales_vmd['Mbps (Media de bajada)'], errors='coerce')

# Paso 2: Manejar valores nulos
# Rellenar nulos en 'Mbps (Media de bajada)' con la mediana o un valor adecuado
df_totales_vmd['Mbps (Media de bajada)'] = df_totales_vmd['Mbps (Media de bajada)'].fillna(df_totales_vmd['Mbps (Media de bajada)'].median())

# Paso 3: Eliminar o manejar valores atípicos en 'Mbps (Media de bajada)'
# Por ejemplo, si la velocidad es mayor a un límite razonable (por ejemplo, 500 Mbps), se puede considerar un valor atípico
df_totales_vmd = df_totales_vmd[df_totales_vmd['Mbps (Media de bajada)'] <= 500]

# Paso 4: Verificar resultados
print("\nInformación después de la limpieza:")
print(df_totales_vmd.info())
print(df_totales_vmd.describe())

# Paso 5: Guardar el DataFrame limpio
df_totales_vmd.to_csv('Totales_VMD_limpio.csv', index=False)


Información inicial del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     42 non-null     int64  
 1   Trimestre               42 non-null     int64  
 2   Mbps (Media de bajada)  42 non-null     float64
 3   Trimestre.1             42 non-null     object 
dtypes: float64(1), int64(2), object(1)
memory usage: 1.4+ KB
None
               Año  Trimestre  Mbps (Media de bajada)
count    42.000000  42.000000               42.000000
mean   2018.761905   2.452381               39.334762
std       3.074756   1.130560               43.643195
min    2014.000000   1.000000                3.617127
25%    2016.000000   1.250000                6.038428
50%    2019.000000   2.000000               18.855000
75%    2021.000000   3.000000               51.370000
max    2024.000000   4.000000              139

Analisis y ETL de la Hoja1: Totales Accesos Por Tecnología

In [9]:
# Totales Accesos Por Tecnología
df_totales_accesos_por_tecnologia = pd.read_excel(archivo_excel, sheet_name='Totales Accesos Por Tecnología')
print("\nHoja: Totales Accesos Por Tecnología")
print(df_totales_accesos_por_tecnologia.head())
print(df_totales_accesos_por_tecnologia.info())
print(df_totales_accesos_por_tecnologia.describe())


Hoja: Totales Accesos Por Tecnología
    Año  Trimestre     ADSL  Cablemodem  Fibra óptica  Wireless   Otros  \
0  2024          2   733491     5867504       4169958    593197  191957   
1  2024          1   774475     5986957       4015101    598682  257941   
2  2023          4   836390     6022532       3908183    585760  194796   
3  2023          3   897895     6018832       3708718    581436  200027   
4  2023          2  1006509     5997149       3463988    581823  202428   

      Total        Periodo  
0  11556107   Abr-Jun 2024  
1  11633156   Ene-Mar 2024  
2  11547661   Oct-Dic 2023  
3  11406908  Jul-Sept 2023  
4  11251897   Abr-Jun 2023  
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Año           42 non-null     int64 
 1   Trimestre     42 non-null     int64 
 2   ADSL          42 non-null     int64 
 3   Cablemodem    42 non

In [24]:
# Cargar la hoja
archivo_excel = 'https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx'  # ruta al archivo
df_totales_accesos = pd.read_excel(archivo_excel, sheet_name='Totales Accesos Por Tecnología')

# Mostrar información inicial
print("Información inicial del DataFrame:")
print(df_totales_accesos.info())
print(df_totales_accesos.describe())

# Paso 1: Asegurarse que las columnas de accesos sean numéricas
columnas_accesos = ['ADSL', 'Cablemodem', 'Fibra óptica', 'Wireless', 'Otros', 'Total']
df_totales_accesos[columnas_accesos] = df_totales_accesos[columnas_accesos].apply(pd.to_numeric, errors='coerce')

# Paso 2: Manejar valores nulos
# Rellenar nulos con la media de cada columna
df_totales_accesos[columnas_accesos] = df_totales_accesos[columnas_accesos].fillna(df_totales_accesos[columnas_accesos].mean())

# Paso 3: Verificar si los totales suman correctamente
df_totales_accesos['Total Calculado'] = df_totales_accesos[columnas_accesos[:-1]].sum(axis=1)

# Comparar Total Calculado con el valor en la columna 'Total'
df_totales_accesos['Total Correcto'] = df_totales_accesos['Total'] == df_totales_accesos['Total Calculado']

# Paso 4: Eliminar la columna 'Total Calculado' si no es necesaria
df_totales_accesos.drop(columns=['Total Calculado'], inplace=True)

# Paso 5: Verificar resultados
print("\nInformación después de la limpieza:")
print(df_totales_accesos.info())
print(df_totales_accesos.describe())

# Paso 6: Guardar el DataFrame limpio
df_totales_accesos.to_csv('Totales_Accecos_Por_Tecnologia_limpio.csv', index=False)


Información inicial del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Año           42 non-null     int64 
 1   Trimestre     42 non-null     int64 
 2   ADSL          42 non-null     int64 
 3   Cablemodem    42 non-null     int64 
 4   Fibra óptica  42 non-null     int64 
 5   Wireless      42 non-null     int64 
 6   Otros         42 non-null     int64 
 7   Total         42 non-null     int64 
 8   Periodo       42 non-null     object
dtypes: int64(8), object(1)
memory usage: 3.1+ KB
None
               Año  Trimestre          ADSL    Cablemodem  Fibra óptica  \
count    42.000000  42.000000  4.200000e+01  4.200000e+01  4.200000e+01   
mean   2018.761905   2.452381  2.694844e+06  4.433906e+06  1.224694e+06   
std       3.074756   1.130560  1.094347e+06  1.319783e+06  1.321136e+06   
min    2014.000000   1.000000  7.334910e+05  2

Analisis y ETL de la Hoja1: Accesos Por Tecnología

In [10]:
# Accesos Por Tecnología
df_accesos_por_tecnologia = pd.read_excel(archivo_excel, sheet_name='Accesos Por Tecnología')
print("\nHoja: Accesos Por Tecnología")
print(df_accesos_por_tecnologia.head())
print(df_accesos_por_tecnologia.info())
print(df_accesos_por_tecnologia.describe())


Hoja: Accesos Por Tecnología
    Año Trimestre        Provincia      ADSL  Cablemodem  Fibra óptica  \
0  2024         2     Buenos Aires  214055.0   2722466.0     1849476.0   
1  2024         2  Capital Federal   54102.0   1144781.0      230402.0   
2  2024         2        Catamarca    4951.0     10303.0       58355.0   
3  2024         2            Chaco    9448.0     57935.0       68944.0   
4  2024         2           Chubut   25955.0     80704.0       26516.0   

   Wireless    Otros      Total  
0  138638.0  64745.0  4989380.0  
1    4493.0  29821.0  1463599.0  
2    1384.0     81.0    75074.0  
3    8407.0   2358.0   147092.0  
4   31118.0   9930.0   174223.0  
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1009 entries, 0 to 1008
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           1008 non-null   object 
 1   Trimestre     1008 non-null   object 
 2   Provincia     1007 non-null   object 


In [25]:
# Cargar la hoja
archivo_excel = 'https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx'  # ruta al archivo
df_accesos_tecnologia = pd.read_excel(archivo_excel, sheet_name='Accesos Por Tecnología')

# Mostrar información inicial
print("Información inicial del DataFrame:")
print(df_accesos_tecnologia.info())
print(df_accesos_tecnologia.describe())

# Paso 1: Asegurarse que las columnas de accesos sean numéricas
columnas_accesos = ['ADSL', 'Cablemodem', 'Fibra óptica', 'Wireless', 'Otros', 'Total']
df_accesos_tecnologia[columnas_accesos] = df_accesos_tecnologia[columnas_accesos].apply(pd.to_numeric, errors='coerce')

# Paso 2: Manejar valores nulos
# Rellenar nulos con la media de cada columna
df_accesos_tecnologia[columnas_accesos] = df_accesos_tecnologia[columnas_accesos].fillna(df_accesos_tecnologia[columnas_accesos].mean())

# Paso 3: Verificar si los totales suman correctamente
df_accesos_tecnologia['Total Calculado'] = df_accesos_tecnologia[columnas_accesos[:-1]].sum(axis=1)

# Comparar Total Calculado con el valor en la columna 'Total'
df_accesos_tecnologia['Total Correcto'] = df_accesos_tecnologia['Total'] == df_accesos_tecnologia['Total Calculado']

# Paso 4: Eliminar la columna 'Total Calculado' si no es necesaria
df_accesos_tecnologia.drop(columns=['Total Calculado'], inplace=True)

# Paso 5: Verificar resultados
print("\nInformación después de la limpieza:")
print(df_accesos_tecnologia.info())
print(df_accesos_tecnologia.describe())

# Paso 6: Guardar el DataFrame limpio
df_accesos_tecnologia.to_csv('Accesos_Por_Tecnologia_limpio.csv', index=False)


Información inicial del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1009 entries, 0 to 1008
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           1008 non-null   object 
 1   Trimestre     1008 non-null   object 
 2   Provincia     1007 non-null   object 
 3   ADSL          1007 non-null   float64
 4   Cablemodem    1007 non-null   float64
 5   Fibra óptica  1007 non-null   float64
 6   Wireless      1007 non-null   float64
 7   Otros         1007 non-null   float64
 8   Total         1007 non-null   float64
dtypes: float64(6), object(3)
memory usage: 71.1+ KB
None
               ADSL    Cablemodem  Fibra óptica       Wireless         Otros  \
count  1.007000e+03  1.007000e+03  1.007000e+03    1007.000000   1007.000000   
mean   1.129309e+05  1.821583e+05  4.855699e+04   12542.229394   6497.041708   
std    2.362922e+05  4.460058e+05  1.758694e+05   20909.412762  11239.855091   
min    2

Analisis y ETL de la Hoja1: Penetración-poblacion

In [11]:
# Penetración-poblacion
df_penetracion_poblacion = pd.read_excel(archivo_excel, sheet_name='Penetración-poblacion')
print("\nHoja: Penetración-poblacion")
print(df_penetracion_poblacion.head())
print(df_penetracion_poblacion.info())
print(df_penetracion_poblacion.describe())


Hoja: Penetración-poblacion
    Año  Trimestre        Provincia  Accesos por cada 100 hab
0  2024          2     Buenos Aires                     27.43
1  2024          2  Capital Federal                     47.44
2  2024          2        Catamarca                     17.50
3  2024          2            Chaco                     11.78
4  2024          2           Chubut                     26.46
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Año                       1000 non-null   int64  
 1   Trimestre                 1000 non-null   int64  
 2   Provincia                 1000 non-null   object 
 3   Accesos por cada 100 hab  1000 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 31.4+ KB
None
               Año    Trimestre  Accesos por cada 100 hab
count  1000.000000  1000.000000       

In [26]:
# Cargar la hoja
archivo_excel = 'https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx'  # ruta al archivo
df_penetracion_poblacion = pd.read_excel(archivo_excel, sheet_name='Penetración-poblacion')

# Mostrar información inicial
print("Información inicial del DataFrame:")
print(df_penetracion_poblacion.info())
print(df_penetracion_poblacion.describe())

# Paso 1: Asegurarse de que la columna 'Accesos por cada 100 hab' sea numérica
df_penetracion_poblacion['Accesos por cada 100 hab'] = pd.to_numeric(df_penetracion_poblacion['Accesos por cada 100 hab'], errors='coerce')

# Paso 2: Verificar si hay valores nulos
if df_penetracion_poblacion['Accesos por cada 100 hab'].isnull().sum() > 0:
    print(f"Hay {df_penetracion_poblacion['Accesos por cada 100 hab'].isnull().sum()} valores nulos en 'Accesos por cada 100 hab'.")
    # Rellenar nulos con la media de la columna
    df_penetracion_poblacion['Accesos por cada 100 hab'] = df_penetracion_poblacion['Accesos por cada 100 hab'].fillna(df_penetracion_poblacion['Accesos por cada 100 hab'].mean())

# Paso 3: Revisar si hay valores duplicados
duplicados = df_penetracion_poblacion.duplicated().sum()
if duplicados > 0:
    print(f"Hay {duplicados} filas duplicadas.")
    df_penetracion_poblacion = df_penetracion_poblacion.drop_duplicates()

# Paso 4: Verificar resultados después de la limpieza
print("\nInformación después de la limpieza:")
print(df_penetracion_poblacion.info())
print(df_penetracion_poblacion.describe())

# Paso 5: Guardar el DataFrame limpio
df_penetracion_poblacion.to_csv('Penetracion_poblacion_limpio.csv', index=False)


Información inicial del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Año                       1000 non-null   int64  
 1   Trimestre                 1000 non-null   int64  
 2   Provincia                 1000 non-null   object 
 3   Accesos por cada 100 hab  1000 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 31.4+ KB
None
               Año    Trimestre  Accesos por cada 100 hab
count  1000.000000  1000.000000               1000.000000
mean   2018.800000     2.464000                 15.727248
std       3.021445     1.114426                  9.240063
min    2014.000000     1.000000                  2.723000
25%    2016.000000     1.000000                  9.018188
50%    2019.000000     2.000000                 13.674812
75%    2021.000000     3.000000                 20.32397

Analisis y ETL de la Hoja1: Penetracion-hogaresos

In [12]:
# Penetracion-hogares
df_penetracion_hogares = pd.read_excel(archivo_excel, sheet_name='Penetracion-hogares')
print("\nHoja: Penetracion-hogares")
print(df_penetracion_hogares.head())
print(df_penetracion_hogares.info())
print(df_penetracion_hogares.describe())


Hoja: Penetracion-hogares
    Año  Trimestre        Provincia  Accesos por cada 100 hogares
0  2024          2     Buenos Aires                         79.84
1  2024          2  Capital Federal                        116.37
2  2024          2        Catamarca                         68.81
3  2024          2            Chaco                         44.06
4  2024          2           Chubut                         86.33
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           1000 non-null   int64  
 1   Trimestre                     1000 non-null   int64  
 2   Provincia                     1000 non-null   object 
 3   Accesos por cada 100 hogares  1000 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 31.4+ KB
None
               Año    Trimestre  Accesos por cada 

In [27]:
# Cargar la hoja
archivo_excel = 'https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx'  # ruta al archivo
df_penetracion_hogares = pd.read_excel(archivo_excel, sheet_name='Penetracion-hogares')

# Mostrar información inicial
print("Información inicial del DataFrame:")
print(df_penetracion_hogares.info())
print(df_penetracion_hogares.describe())

# Paso 1: Asegurarse de que la columna 'Accesos por cada 100 hogares' sea numérica
df_penetracion_hogares['Accesos por cada 100 hogares'] = pd.to_numeric(df_penetracion_hogares['Accesos por cada 100 hogares'], errors='coerce')

# Paso 2: Verificar si hay valores nulos
if df_penetracion_hogares['Accesos por cada 100 hogares'].isnull().sum() > 0:
    print(f"Hay {df_penetracion_hogares['Accesos por cada 100 hogares'].isnull().sum()} valores nulos en 'Accesos por cada 100 hogares'.")
    # Rellenar nulos con la media de la columna
    df_penetracion_hogares['Accesos por cada 100 hogares'] = df_penetracion_hogares['Accesos por cada 100 hogares'].fillna(df_penetracion_hogares['Accesos por cada 100 hogares'].mean())

# Paso 3: Revisar si hay valores duplicados
duplicados = df_penetracion_hogares.duplicated().sum()
if duplicados > 0:
    print(f"Hay {duplicados} filas duplicadas.")
    df_penetracion_hogares = df_penetracion_hogares.drop_duplicates()

# Paso 4: Verificar resultados después de la limpieza
print("\nInformación después de la limpieza:")
print(df_penetracion_hogares.info())
print(df_penetracion_hogares.describe())

# Paso 5: Guardar el DataFrame limpio
df_penetracion_hogares.to_csv('Penetracion_hogares_limpio.csv', index=False)


Información inicial del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           1000 non-null   int64  
 1   Trimestre                     1000 non-null   int64  
 2   Provincia                     1000 non-null   object 
 3   Accesos por cada 100 hogares  1000 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 31.4+ KB
None
               Año    Trimestre  Accesos por cada 100 hogares
count  1000.000000  1000.000000                   1000.000000
mean   2018.800000     2.464000                     52.794973
std       3.021445     1.114426                     24.135063
min    2014.000000     1.000000                      9.513970
25%    2016.000000     1.000000                     33.957657
50%    2019.000000     2.000000                     50.133004
75% 

Analisis y ETL de la Hoja1: Penetracion-totales

In [13]:
# Penetracion-totales
df_penetracion_totales = pd.read_excel(archivo_excel, sheet_name='Penetracion-totales')
print("\nHoja: Penetracion-totales")
print(df_penetracion_totales.head())
print(df_penetracion_totales.info())
print(df_penetracion_totales.describe())


Hoja: Penetracion-totales
    Año  Trimestre  Accesos por cada 100 hogares  Accesos por cada 100 hab  \
0  2024          2                         78.13                     24.57   
1  2024          1                         78.89                     24.79   
2  2023          4                         78.56                     24.66   
3  2023          3                         77.84                     24.41   
4  2023          2                         77.02                     24.14   

         Periodo  
0   Abr-Jun 2024  
1   Ene-Mar 2024  
2   Oct-Dic 2023  
3  Jul-Sept 2023  
4   Abr-Jun 2023  
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           42 non-null     int64  
 1   Trimestre                     42 non-null     int64  
 2   Accesos por cada 100 hogares  42 non-null     f

In [28]:
# Cargar la hoja
archivo_excel = 'https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx'  # ruta al archivo
df_penetracion_totales = pd.read_excel(archivo_excel, sheet_name='Penetracion-totales')

# Mostrar información inicial
print("Información inicial del DataFrame:")
print(df_penetracion_totales.info())
print(df_penetracion_totales.describe())

# Paso 1: Asegurarse de que las columnas numéricas sean del tipo correcto
df_penetracion_totales['Accesos por cada 100 hogares'] = pd.to_numeric(df_penetracion_totales['Accesos por cada 100 hogares'], errors='coerce')
df_penetracion_totales['Accesos por cada 100 hab'] = pd.to_numeric(df_penetracion_totales['Accesos por cada 100 hab'], errors='coerce')

# Paso 2: Verificar si hay valores nulos
if df_penetracion_totales.isnull().sum().sum() > 0:
    print(f"Hay valores nulos en el DataFrame.")
    # Rellenar nulos con la media de cada columna
    df_penetracion_totales = df_penetracion_totales.fillna(df_penetracion_totales.mean())

# Paso 3: Revisar si hay valores duplicados
duplicados = df_penetracion_totales.duplicated().sum()
if duplicados > 0:
    print(f"Hay {duplicados} filas duplicadas.")
    df_penetracion_totales = df_penetracion_totales.drop_duplicates()

# Paso 4: Verificar resultados después de la limpieza
print("\nInformación después de la limpieza:")
print(df_penetracion_totales.info())
print(df_penetracion_totales.describe())

# Paso 5: Guardar el DataFrame limpio
df_penetracion_totales.to_csv('Penetracion_totales_limpio.csv', index=False)


Información inicial del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           42 non-null     int64  
 1   Trimestre                     42 non-null     int64  
 2   Accesos por cada 100 hogares  42 non-null     float64
 3   Accesos por cada 100 hab      42 non-null     float64
 4   Periodo                       42 non-null     object 
dtypes: float64(2), int64(2), object(1)
memory usage: 1.8+ KB
None
               Año  Trimestre  Accesos por cada 100 hogares  \
count    42.000000  42.000000                     42.000000   
mean   2018.761905   2.452381                     63.201813   
std       3.074756   1.130560                      9.950321   
min    2014.000000   1.000000                     49.545821   
25%    2016.000000   1.250000                     53.784644   
50%  

Analisis y ETL de la Hoja1: Totales Accesos por rango

In [14]:
# Totales Accesos por rango
df_totales_accesos_por_rango = pd.read_excel(archivo_excel, sheet_name='Totales Accesos por rango')
print("\nHoja: Totales Accesos por rango")
print(df_totales_accesos_por_rango.head())
print(df_totales_accesos_por_rango.info())
print(df_totales_accesos_por_rango.describe())


Hoja: Totales Accesos por rango
    Año  Trimestre  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
0  2024          2           28151                    67024   
1  2024          1           28801                    69355   
2  2023          4           29708                    71742   
3  2023          3           30274                    80609   
4  2023          2           30827                    79405   

   Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  Entre 10 Mbps y 20 Mbps  \
0                 840200                  911374                   662649   
1                 866152                  950930                   672155   
2                 900253                  978108                   697232   
3                 945608                  998725                   720084   
4                 963510                 1028250                   734125   

   Entre 20 Mbps y 30 Mbps  Más de 30 Mbps   OTROS     Total  
0                   348253         8357088  341368  11556107  

In [30]:
# Cargar la hoja
archivo_excel = 'https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx'  # ruta al archivo
df_accesos_por_rango = pd.read_excel(archivo_excel, sheet_name='Totales Accesos por rango')

# Mostrar información inicial
print("Información inicial del DataFrame:")
print(df_accesos_por_rango.info())
print(df_accesos_por_rango.describe())

# Paso 1: Asegurarse de que las columnas sean del tipo correcto
df_accesos_por_rango = df_accesos_por_rango.apply(pd.to_numeric, errors='coerce')

# Paso 2: Verificar si hay valores nulos
if df_accesos_por_rango.isnull().sum().sum() > 0:
    print(f"Hay valores nulos en el DataFrame.")
    # Rellenar nulos con la media de cada columna
    df_accesos_por_rango = df_accesos_por_rango.fillna(df_accesos_por_rango.mean())

# Paso 3: Revisar si hay valores duplicados
duplicados = df_accesos_por_rango.duplicated().sum()
if duplicados > 0:
    print(f"Hay {duplicados} filas duplicadas.")
    df_accesos_por_rango = df_accesos_por_rango.drop_duplicates()

# Paso 4: Verificar que la columna Total sea la suma de las otras columnas
df_accesos_por_rango['Total_calculado'] = df_accesos_por_rango.iloc[:, 2:10].sum(axis=1)
df_accesos_por_rango['Total_correcto'] = df_accesos_por_rango['Total'] == df_accesos_por_rango['Total_calculado']
print(f"Filas donde Total no coincide: {df_accesos_por_rango[df_accesos_por_rango['Total_correcto'] == False]}")

# Paso 5: Verificar resultados después de la limpieza
print("\nInformación después de la limpieza:")
print(df_accesos_por_rango.info())
print(df_accesos_por_rango.describe())

# Paso 6: Guardar el DataFrame limpio
df_accesos_por_rango.to_csv('Totales_Accesos_por_rango_limpio.csv', index=False)


Información inicial del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 11 columns):
 #   Column                   Non-Null Count  Dtype
---  ------                   --------------  -----
 0   Año                      42 non-null     int64
 1   Trimestre                42 non-null     int64
 2   Hasta 512 kbps           42 non-null     int64
 3   Entre 512 Kbps y 1 Mbps  42 non-null     int64
 4   Entre 1 Mbps y 6 Mbps    42 non-null     int64
 5   Entre 6 Mbps y 10 Mbps   42 non-null     int64
 6   Entre 10 Mbps y 20 Mbps  42 non-null     int64
 7   Entre 20 Mbps y 30 Mbps  42 non-null     int64
 8   Más de 30 Mbps           42 non-null     int64
 9   OTROS                    42 non-null     int64
 10  Total                    42 non-null     int64
dtypes: int64(11)
memory usage: 3.7 KB
None
               Año  Trimestre  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
count    42.000000  42.000000       42.000000                42.0000

Analisis y ETL de la Hoja1: Accesos por rangos

In [15]:
# Accesos por rangos
df_accesos_por_rangos = pd.read_excel(archivo_excel, sheet_name='Accesos por rangos')
print("\nHoja: Accesos por rangos")
print(df_accesos_por_rangos.head())
print(df_accesos_por_rangos.info())
print(df_accesos_por_rangos.describe())


Hoja: Accesos por rangos
    Año  Trimestre        Provincia  HASTA 512 kbps  + 512 Kbps - 1 Mbps  \
0  2024          2     Buenos Aires         25287.0              23034.0   
1  2024          2  Capital Federal           518.0               4122.0   
2  2024          2        Catamarca            72.0                164.0   
3  2024          2            Chaco           236.0                349.0   
4  2024          2           Chubut           125.0               1139.0   

   + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  + 10 Mbps - 20 Mbps  \
0           222627.0            221660.0             244528.0   
1            24539.0             40296.0              33158.0   
2             2660.0              3093.0               3799.0   
3            14594.0              8336.0               6208.0   
4            43699.0             31971.0              22696.0   

   + 20 Mbps - 30 Mbps  + 30 Mbps     OTROS      Total  
0             106302.0  4020395.0  125547.0  4989380.0  
1           

In [31]:
# 1. Extracción: Cargar el archivo de Excel
df = pd.read_excel('https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx', sheet_name='Accesos por rangos')

# 2. Transformación:
# Renombrar columnas para mayor legibilidad
df.columns = df.columns.str.strip().str.replace(' ', '_')

# Rellenar los valores nulos en la columna 'OTROS' con 0
df['OTROS'] = df['OTROS'].fillna(0)

# Verificar que los tipos de datos sean correctos
df.dtypes

# Comprobar la consistencia entre el total y las sumas de los rangos
df['Suma_Rangos'] = (df['HASTA_512_kbps'] +
                     df['+_512_Kbps_-_1_Mbps'] +
                     df['+_1_Mbps_-_6_Mbps'] +
                     df['+_6_Mbps_-_10_Mbps'] +
                     df['+_10_Mbps_-_20_Mbps'] +
                     df['+_20_Mbps_-_30_Mbps'] +
                     df['+_30_Mbps'] +
                     df['OTROS'])

# Verificar si la suma de los rangos es igual al total
df['Consistencia'] = df['Suma_Rangos'] == df['Total']

# Mostrar las filas con inconsistencias
inconsistencias = df[df['Consistencia'] == False]
if not inconsistencias.empty:
    print("Inconsistencias encontradas:")
    print(inconsistencias[['Suma_Rangos', 'Total', 'Consistencia']])
else:
    print("No se encontraron inconsistencias en la suma de los rangos.")

# 3. Carga: Exportar el DataFrame limpio

# Guardar como CSV
df.to_csv('accesos_por_rangos_limpio.csv', index=False)

# Verificación final
# Verificar las primeras filas del DataFrame limpio
print("Primeras filas del DataFrame limpio:")
print(df.head())

# Verificar si hay valores nulos restantes
print("Valores nulos por columna:")
print(df.isnull().sum())

Inconsistencias encontradas:
     Suma_Rangos       Total  Consistencia
697   1262939.84  1262939.84         False
745   1261685.84  1261685.84         False
817   1350449.76  1350449.76         False
840   2856077.76  2856077.76         False
889   1333455.08  1333455.08         False
912   2705953.92  2705953.92         False
Primeras filas del DataFrame limpio:
    Año  Trimestre        Provincia  HASTA_512_kbps  +_512_Kbps_-_1_Mbps  \
0  2024          2     Buenos Aires         25287.0              23034.0   
1  2024          2  Capital Federal           518.0               4122.0   
2  2024          2        Catamarca            72.0                164.0   
3  2024          2            Chaco           236.0                349.0   
4  2024          2           Chubut           125.0               1139.0   

   +_1_Mbps_-_6_Mbps  +_6_Mbps_-_10_Mbps  +_10_Mbps_-_20_Mbps  \
0           222627.0            221660.0             244528.0   
1            24539.0             40296.0      

In [32]:
# Redondear a 2 decimales para evitar problemas de redondeo
df['Suma_Rangos'] = df['Suma_Rangos'].round(2)
df['Total'] = df['Total'].round(2)

# Verificar la consistencia después del redondeo
df['Consistencia'] = df['Suma_Rangos'] == df['Total']


In [33]:
# Filtrar las filas con inconsistencias
inconsistencias = df[df['Consistencia'] == False]

if not inconsistencias.empty:
    print("Inconsistencias encontradas:")
    print(inconsistencias[['Suma_Rangos', 'Total', 'Consistencia']])
else:
    print("No se encontraron inconsistencias en la suma de los rangos.")

No se encontraron inconsistencias en la suma de los rangos.


Analisis y ETL de la Hoja1: Dial-BAf

In [16]:
# Dial-BAf
df_dial_baf = pd.read_excel(archivo_excel, sheet_name='Dial-BAf')
print("\nHoja: Dial-BAf")
print(df_dial_baf.head())
print(df_dial_baf.info())
print(df_dial_baf.describe())


Hoja: Dial-BAf
    Año  Trimestre        Provincia  Banda ancha fija  Dial up    Total
0  2024          2     Buenos Aires           4983480   5900.0  4989380
1  2024          2  Capital Federal           1461549   2050.0  1463599
2  2024          2        Catamarca             75073      1.0    75074
3  2024          2            Chaco            147087      5.0   147092
4  2024          2           Chubut            173366    857.0   174223
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1992 entries, 0 to 1991
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Año               1992 non-null   int64  
 1   Trimestre         1992 non-null   int64  
 2   Provincia         1992 non-null   object 
 3   Banda ancha fija  1992 non-null   int64  
 4   Dial up           1988 non-null   float64
 5   Total             1992 non-null   int64  
dtypes: float64(1), int64(4), object(1)
memory usage: 93.5+ KB
None
  

In [37]:
# Cargar el archivo Excel y la hoja específica
archivo = 'https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx'  # ruta real del archivo
df = pd.read_excel(archivo, sheet_name='Dial-BAf')

# Mostrar información inicial del DataFrame
print("Información inicial del DataFrame:")
print(df.info())
print("\nResumen estadístico del DataFrame:")
print(df.describe())

# Paso 1: Limpieza de valores nulos en la columna 'Dial up'
# Rellenar valores nulos con 0 para simplificar la consistencia de la suma
df['Dial up'].fillna(0, inplace=True)

# Paso 2: Verificación de consistencia en la columna 'Total'
# Crear una columna de suma de "Banda ancha fija" y "Dial up" para comparar con "Total"
df['Suma_Rangos'] = df['Banda ancha fija'] + df['Dial up']

# Paso 3: Crear columna de consistencia para comparar "Total" y "Suma_Rangos"
df['Consistencia'] = np.isclose(df['Suma_Rangos'], df['Total'])

# Filtrar las filas donde la consistencia es falsa (inconsistencias)
inconsistencias = df[df['Consistencia'] == False]

# Paso 4: Mostrar resultados finales
print("\nInconsistencias encontradas:")
print(inconsistencias[['Suma_Rangos', 'Total', 'Consistencia']])

# Mostrar las primeras filas del DataFrame limpio y sin inconsistencias
print("\nPrimeras filas del DataFrame limpio:")
print(df.head())

# Resumen de valores nulos por columna
print("\nValores nulos por columna:")
print(df.isnull().sum())


Información inicial del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1992 entries, 0 to 1991
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Año               1992 non-null   int64  
 1   Trimestre         1992 non-null   int64  
 2   Provincia         1992 non-null   object 
 3   Banda ancha fija  1992 non-null   int64  
 4   Dial up           1988 non-null   float64
 5   Total             1992 non-null   int64  
dtypes: float64(1), int64(4), object(1)
memory usage: 93.5+ KB
None

Resumen estadístico del DataFrame:
               Año    Trimestre  Banda ancha fija       Dial up         Total
count  1992.000000  1992.000000      1.992000e+03   1988.000000  1.992000e+03
mean   2018.698795     2.457831      3.650542e+05    902.104125  3.659545e+05
std       3.001697     1.122899      7.858852e+05   2281.691033  7.876088e+05
min    2014.000000     1.000000      1.219300e+04      0.000000  1.

/tmp/ipykernel_4415/2089616333.py:13: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Dial up'].fillna(0, inplace=True)


In [38]:
# Guardar como CSV
df.to_csv('Dial_BAf_limpio.csv', index=False)

Analisis y ETL de la Hoja1: Totales Dial-BAf

In [15]:
# Totales Dial-BAf
df_dial_baf = pd.read_excel(archivo_excel, sheet_name='Totales Dial-BAf')
print("\nHoja: Totales Dial-BAf")
print(df_dial_baf.head())
print(df_dial_baf.info())
print(df_dial_baf.describe())


Hoja: Totales Dial-BAf
    Año  Trimestre  Banda ancha fija  Dial up     Total        Periodo
0  2024          2          11544314    11793  11556107   Abr-Jun 2024
1  2024          1          11621363    11793  11633156   Ene-Mar 2024
2  2023          4          11535868    11793  11547661   Oct-Dic 2023
3  2023          3          11395115    11793  11406908  Jul-Sept 2023
4  2023          2          11239781    12116  11251897   Abr-Jun 2023
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Año               42 non-null     int64 
 1   Trimestre         42 non-null     int64 
 2   Banda ancha fija  42 non-null     int64 
 3   Dial up           42 non-null     int64 
 4   Total             42 non-null     int64 
 5   Periodo           42 non-null     object
dtypes: int64(5), object(1)
memory usage: 2.1+ KB
None
               Año  Trime

In [16]:
# Cargar el archivo Excel (ajusta el nombre y la ruta del archivo según sea necesario)
archivo_excel = "https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx"
hoja = "Totales Dial-BAf"

# Cargar la hoja en un DataFrame
df = pd.read_excel(archivo_excel, sheet_name=hoja)

# 1. Inspección inicial
print("Vista previa de los datos:")
print(df.head())
print("\nInformación general:")
print(df.info())
print("\nEstadísticas descriptivas:")
print(df.describe())

# 2. Limpieza de datos
# Verificar si hay valores nulos
print("\nValores nulos por columna:")
print(df.isnull().sum())

# Si hubiera valores nulos, podríamos tratarlos (en este caso no hay, pero dejo el ejemplo)
# df['Dial up'].fillna(0, inplace=True)  # Ejemplo: Rellenar con 0

# Normalizar los nombres de las columnas
df.columns = [col.strip().replace(" ", "_").lower() for col in df.columns]

# 3. Transformación de datos
# Convertir la columna "periodo" en formato datetime (opcional si es relevante)
df['periodo'] = pd.to_datetime(df['periodo'], errors='coerce', format='%b-%Y')

# Agregar columnas calculadas (ejemplo: proporción de "Dial up" respecto al total)
df['proporcion_dial_up'] = df['dial_up'] / df['total']

# 4. Validación de datos
print("\nValidación de datos transformados:")
print(df.head())

df.to_csv('Totales Dial_BAf_limpio.csv', index=False)

Vista previa de los datos:
    Año  Trimestre  Banda ancha fija  Dial up     Total        Periodo
0  2024          2          11544314    11793  11556107   Abr-Jun 2024
1  2024          1          11621363    11793  11633156   Ene-Mar 2024
2  2023          4          11535868    11793  11547661   Oct-Dic 2023
3  2023          3          11395115    11793  11406908  Jul-Sept 2023
4  2023          2          11239781    12116  11251897   Abr-Jun 2023

Información general:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Año               42 non-null     int64 
 1   Trimestre         42 non-null     int64 
 2   Banda ancha fija  42 non-null     int64 
 3   Dial up           42 non-null     int64 
 4   Total             42 non-null     int64 
 5   Periodo           42 non-null     object
dtypes: int64(5), object(1)
memory usage: 2.1+ KB
None


Analisis y ETL de la Hoja1: Ingresos

In [45]:
# Ingresos
df_ingresos = pd.read_excel(archivo_excel, sheet_name='Ingresos ')
print("\nHoja: Ingresos")
print(df_ingresos.head())
print(df_ingresos.info())
print(df_ingresos.describe())


Hoja: Ingresos
    Año  Trimestre  Ingresos (miles de pesos)        Periodo
0  2024          2               4.420322e+08   Abr-Jun 2024
1  2024          1               3.461990e+08   Ene-Mar 2024
2  2023          4               1.673760e+08   Oct-Dic 2023
3  2023          3               1.331066e+08  Jul-Sept 2023
4  2023          2               1.180603e+08   Jun-Mar 2023
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 4 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        42 non-null     int64  
 1   Trimestre                  42 non-null     int64  
 2   Ingresos (miles de pesos)  42 non-null     float64
 3   Periodo                    42 non-null     object 
dtypes: float64(1), int64(2), object(1)
memory usage: 1.4+ KB
None
               Año  Trimestre  Ingresos (miles de pesos)
count    42.000000  42.000000               4.200000e+01

In [54]:
# Cargar el archivo Excel
file_path = 'https://indicadores.enacom.gob.ar/Files/Datos_Abiertos/Internet.xlsx'
df_ingresos = pd.read_excel(file_path, sheet_name='Ingresos ')

# Verificar los primeros registros
print(df_ingresos.head())

    Año  Trimestre  Ingresos (miles de pesos)        Periodo
0  2024          2               4.420322e+08   Abr-Jun 2024
1  2024          1               3.461990e+08   Ene-Mar 2024
2  2023          4               1.673760e+08   Oct-Dic 2023
3  2023          3               1.331066e+08  Jul-Sept 2023
4  2023          2               1.180603e+08   Jun-Mar 2023


In [55]:
# Normalización de nombres de las columnas
df_ingresos.columns = df_ingresos.columns.str.strip()

# Convertir el periodo en fechas. Suponiendo que cada trimestre comienza en el primer día del mes del trimestre
# Esto es un ejemplo de cómo podrías hacerlo.
def obtener_fecha_inicio_trimestre(periodo):
    trimestres = {
        'Ene-Mar': '01-01', 'Abr-Jun': '04-01', 'Jul-Sept': '07-01', 'Oct-Dic': '10-01'
    }
    return f"{periodo.split(' ')[-1]}-{trimestres.get(periodo.split(' ')[0], '01-01')}"

# Crear una columna con la fecha de inicio del periodo
df_ingresos['Fecha de inicio'] = df_ingresos['Periodo'].apply(obtener_fecha_inicio_trimestre)

# Convertir a formato fecha
df_ingresos['Fecha de inicio'] = pd.to_datetime(df_ingresos['Fecha de inicio'], format='%Y-%m-%d')

# Verificamos los resultados
print(df_ingresos.head())


    Año  Trimestre  Ingresos (miles de pesos)        Periodo Fecha de inicio
0  2024          2               4.420322e+08   Abr-Jun 2024      2024-04-01
1  2024          1               3.461990e+08   Ene-Mar 2024      2024-01-01
2  2023          4               1.673760e+08   Oct-Dic 2023      2023-10-01
3  2023          3               1.331066e+08  Jul-Sept 2023      2023-07-01
4  2023          2               1.180603e+08   Jun-Mar 2023      2023-01-01


In [56]:
df.to_csv('Ingresos_limpio.csv', index=False)

HOJAS ELEGIDAS

1. Análisis: se divide en las siguientes áreas clave:

1.1 Penetración del Servicio
Objetivo: Evaluar qué tan extendido está el servicio de internet en la población y en los hogares.
Hojas relevantes:
- Penetración-poblacion
- Penetracion-hogares
- Penetracion-totales

Preguntas clave:
¿Qué porcentaje de la población tiene acceso a internet en cada provincia?
¿Cómo varía la penetración de internet en los hogares según la región?
¿Cuáles son las tendencias en la penetración total a lo largo del tiempo?


1.2 Calidad y Velocidad del Servicio
Objetivo: Identificar las velocidades promedio y su distribución entre provincias.
Hojas relevantes:
- Velocidad_sin_Rangos
- Velocidad % por prov
- Totales VMD
- Accesos por rangos
- Totales Accesos por rango

Preguntas clave:
¿Qué velocidades son más comunes y cómo se distribuyen geográficamente?
¿Cuál es la velocidad promedio de descarga en cada región?
¿Existen disparidades significativas en la calidad del servicio entre provincias?


1.3 Tecnologías de Conexión
Objetivo: Analizar las tecnologías dominantes y su evolución.
Hojas relevantes:
- Accesos_tecnologia_localidad
- Totales Accesos Por Tecnología
- Accesos Por Tecnología

Preguntas clave:
¿Qué tecnologías de conexión son las más utilizadas?
¿Cómo ha cambiado el uso de tecnologías como fibra óptica o ADSL a lo largo del tiempo?
¿Qué provincias tienen mayor adopción de tecnologías modernas?